azure/packages/azure-client/src/test/AzureClient.spec.ts

```diff
@@ -45,239 +45,614 @@ describe("AzureClient", () => {
+            assert.strictEqual(
+                container.attachState,
+                AttachState.Attached,
+                "Container is not attached after attach is called",
+            );
+
+            /* This is a workaround for a known bug, we should have one member (self) upon container connection */
+            if (services.audience.getMembers().size === 0) {
+                await new Promise<void>((resolve) => {
+                    services.audience.on("memberAdded", () => {
+                        resolve();
+                    });
+                });
+            }
+
+            const members = services.audience.getMembers();
+            assert.strictEqual(members.size, 1, "We should have only one member at this point.");
+
+            const myself = services.audience.getMyself();
+            assert.notStrictEqual(myself, undefined, "We should have myself at this point.");
         });
 
-        const resources = client.copyContainer(containerId, schema);
-        await assert.doesNotReject(
-            resources,
-            () => true,
-            "container cannot be retrieved from Azure Fluid Relay",
-        );
+        /**
+         * Scenario: Find partner member
+         *
+         * Expected behavior: upon resolving container, the partner member should be able
+         * to resolve original member.
+         */
+        it("can find partner member", async () => {
+            const { container, services } = await client.createContainer(schema);
+            const containerId = await container.attach();
+            await new Promise<void>((resolve) => {
+                container.on("connected", () => {
+                    resolve();
+                });
+            });
+
+            assert.strictEqual(typeof containerId, "string", "Attach did not return a string ID");
+            assert.strictEqual(
+                container.attachState,
+                AttachState.Attached,
+                "Container is not attached after attach is called",
+            );
+
+            /* This is a workaround for a known bug, we should have one member (self) upon container connection */
+            if (services.audience.getMembers().size === 0) {
+                await new Promise<void>((resolve) => {
+                    services.audience.on("memberAdded", () => {
+                        resolve();
+                    });
+                });
+            }
+
+            const originalSelf = services.audience.getMyself();
+            assert.notStrictEqual(originalSelf, undefined, "We should have myself at this point.");
+
+            const client2 = createAzureClient("test-id-2", "test-user-name-2");
+            const { services: servicesGet } = await client2.getContainer(containerId, schema);
+
+            const members = servicesGet.audience.getMembers();
+            assert.strictEqual(members.size, 2, "We should have two members at this point.");
```

Comment:
This doesn't need to wait for myself?

Reply:
Interestingly no. Was planning to point it out to you earlier. It seems like we have different logic play out during getContainer. I ran these tests against FRS as well, just to be sure,. I'll dig in further, but at least we should see test failures on our pipelines, if that bug is valid for "get" flow as well.